src/vs/workbench/contrib/notebook/browser/controller/executeActions.ts

```diff
@@ -730,3 +731,52 @@ registerAction2(class RevealLastFailedCellAction extends NotebookAction {
 		}
 	}
 });
+
+registerAction2(class RevealRecentlyRunCellAction extends NotebookAction {
+	constructor() {
+		super({
+			id: REVEAL_RECENTLY_RUN_CELL,
+			title: localize('revealRecentlyRunCell', "Go to Most Recently Run Cell"),
+			tooltip: localize('revealRecentlyRunCell', "Go to Most Recently Run Cell"),
+			shortTitle: localize('revealRecentlyRunCellShort', "Go To"),
+			precondition: NOTEBOOK_LAST_CELL_FAILED.toNegated() && NOTEBOOK_HAS_RUNNING_CELL.toNegated(),
+			menu: [
+				{
+					id: MenuId.EditorTitle,
+					when: ContextKeyExpr.and(
+						NOTEBOOK_IS_ACTIVE_EDITOR,
+						NOTEBOOK_LAST_CELL_FAILED.toNegated(),
+						NOTEBOOK_HAS_RUNNING_CELL.toNegated(),
+						ContextKeyExpr.notEquals('config.notebook.globalToolbar', true)
+					),
+					group: 'navigation',
+					order: 0
+				},
+				{
+					id: MenuId.NotebookToolbar,
+					when: ContextKeyExpr.and(
+						NOTEBOOK_IS_ACTIVE_EDITOR,
+						NOTEBOOK_LAST_CELL_FAILED.toNegated(),
+						NOTEBOOK_HAS_RUNNING_CELL.toNegated(),
+						ContextKeyExpr.equals('config.notebook.globalToolbar', true)
+					),
+					group: 'navigation/execute',
+					order: 10
+				},
+			],
+			icon: icons.gotoRunCellIcon,
+		});
+	}
+
+	async runWithContext(accessor: ServicesAccessor, context: INotebookActionContext): Promise<void> {
+		const notebookExecutionStateService = accessor.get(INotebookExecutionStateService);
+		const notebook = context.notebookEditor.textModel.uri;
+		const lastSuccessfulCellHandle = notebookExecutionStateService.getLastRunCellForNotebook(notebook);
+		if (lastSuccessfulCellHandle !== undefined) {
```

Comment:
if there is non-last-successfull-cell from the execution service, we can probably use the cell with largetst cell execution count.